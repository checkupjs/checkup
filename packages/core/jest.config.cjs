module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'json'],
  transform: { '\\.ts$': 'ts-jest' },
  coverageReporters: ['lcov', 'text-summary'],
  // collectCoverage: !!process.env.CI,
  collectCoverageFrom: ['src/**/*.ts'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
  snapshotFormat: {
    printBasicPrototype: false,
  },
  testPathIgnorePatterns: ['/__fixtures__/', '/__utils__/'],
  extensionsToTreatAsEsm: ['.ts'],
  globals: {
    'ts-jest': {
      useESM: true,
    },
  },
  moduleNameMapper: {
    '^(\\.{1,2}/.*)\\.js$': '$1',
  },
};
