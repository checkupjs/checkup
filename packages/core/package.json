{
  "name": "@checkup/core",
  "description": "Checkup's core library",
  "version": "0.0.0",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {},
  "devDependencies": {
    "@types/chai": "^4",
    "@types/jest": "^25.1.1",
    "@types/node": "^13",
    "chai": "^4.2.0",
    "jest": "^25.1.0",
    "stdout-stderr": "^0.1.13",
    "ts-jest": "^25.2.0",
    "ts-node": "^8",
    "type-fest": "^0.10.0",
    "typescript": "^3.3"
  },
  "engines": {
    "node": "10.* || >= 12.*"
  },
  "files": [
    "/lib"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc",
    "build:watch": "yarn build -w",
    "test": "jest"
  },
  "types": "lib/index.d.ts"
}
