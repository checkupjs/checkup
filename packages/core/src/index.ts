export { default as BaseTask } from './base-task';
export { default as BaseTaskResult } from './base-task-result';
export { default as FileSearcherTask } from './file-searcher-task';
export { default as FileSearcher } from './searchers/file-searcher';
export { default as CardData } from './pdf-components/card-data';
export { loadPlugins } from './loaders/plugin-loader';
export { getPackageJson } from './utils/get-package-json';
export { default as CheckupConfigService } from './configuration/checkup-config-service';
export { default as getSearchLoader } from './configuration/loaders/get-search-loader';
export { default as getFilepathLoader } from './configuration/loaders/get-filepath-loader';
export { ui } from './utils/ui';
export { toPairs } from './utils/data-transformers';

export * from './types';
