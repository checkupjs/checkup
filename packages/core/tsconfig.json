{
  "compilerOptions": {
    "declaration": true,
    "composite": true,
    "importHelpers": true,
    "module": "commonjs",
    "outDir": "lib",
    "rootDir": "src",
    "strict": true,
    "target": "es2017",
    "types": ["chai", "jest", "node"]
  },
  "include": ["src/**/*"]
}
