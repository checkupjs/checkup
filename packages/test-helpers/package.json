{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
