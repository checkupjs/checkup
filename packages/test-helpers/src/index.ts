export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
