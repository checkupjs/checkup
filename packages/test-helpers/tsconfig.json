{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
