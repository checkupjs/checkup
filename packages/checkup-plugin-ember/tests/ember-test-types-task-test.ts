import '@microsoft/jest-sarif';
import { EmberProject, getTaskContext } from '@checkup/test-helpers';
import { getPluginName } from '@checkup/core';
import EmberTestTypesTask from '../src/tasks/ember-test-types-task';

const TESTS = {
  application: {
    tests: {
      'my-component-test.js': `
      module('test', function(hooks) {
        setupApplicationTest(hooks);

        test('testing foo');
        skip('testing shmoo')
        skip('testing shmoo')
      });
    `,
    },
    someFolder: {
      'my-component-test.js': `
      module('test', function(hooks) {
        setupApplicationTest(hooks);

        test('testing foo');
        skip('testing shmoo')
        skip('testing shmoo')
      });
    `,
    },
  },
  rendering: {
    tests: {
      'my-component-test.js': `
      module('test', function(hooks) {
        setupRenderingTest(hooks);

        test('testing foo');
        skip('testing shmoo')
        skip('testing shmoo')
      });
    `,
    },
  },
  unit: {
    tests: {
      'my-component-test.js': `
      module('test', function(hooks) {
        setupTest(hooks);

        test('testing foo');
        skip('testing shmoo')
        skip('testing shmoo')
      });
    `,
    },
  },
};

describe('ember-test-types-task', () => {
  let project: EmberProject;
  let pluginName = getPluginName(import.meta.url);

  beforeEach(function () {
    project = new EmberProject('checkup-app', '0.0.0');
  });

  afterEach(function () {
    project.dispose();
  });

  it('only renders todos/only test types if they a value > 0', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: {
        rendering: {
          tests: {
            'my-component-test.js': `
              module('test', function(hooks) {
                setupRenderingTest(hooks);

                test('testing foo');
                only('testing shmoo')
                todo('testing shmoo')
              });
            `,
          },
        },
      },
    });

    project.writeSync();

    const results = await new EmberTestTypesTask(
      pluginName,
      getTaskContext({ options: { cwd: project.baseDir }, paths: project.filePaths })
    ).run();

    for (let result of results) {
      expect(result).toBeValidSarifFor('result');
    }
  });

  it('returns all the test types found in the app and outputs to json', async () => {
    project.files = Object.assign(project.files, {
      'index.js': 'index js file',
      addon: TESTS,
    });

    project.writeSync();

    const results = await new EmberTestTypesTask(
      pluginName,
      getTaskContext({ options: { cwd: project.baseDir }, paths: project.filePaths })
    ).run();

    for (let result of results) {
      expect(result).toBeValidSarifFor('result');
    }
  });
});
