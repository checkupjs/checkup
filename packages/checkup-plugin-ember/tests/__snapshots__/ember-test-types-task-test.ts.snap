// Vitest Snapshot v1

exports[`ember-test-types-task > returns all the test types found in the app and outputs to json 1`] = `
[
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/application/someFolder/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 5,
            "startColumn": 9,
            "startLine": 5,
          },
        },
      },
    ],
    "message": {
      "text": "application test found using test",
    },
    "properties": {
      "testMethod": "test",
      "testType": "application",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/application/someFolder/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 6,
            "startColumn": 9,
            "startLine": 6,
          },
        },
      },
    ],
    "message": {
      "text": "application test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "application",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/application/someFolder/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 7,
            "startColumn": 9,
            "startLine": 7,
          },
        },
      },
    ],
    "message": {
      "text": "application test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "application",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/application/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 5,
            "startColumn": 9,
            "startLine": 5,
          },
        },
      },
    ],
    "message": {
      "text": "application test found using test",
    },
    "properties": {
      "testMethod": "test",
      "testType": "application",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/application/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 6,
            "startColumn": 9,
            "startLine": 6,
          },
        },
      },
    ],
    "message": {
      "text": "application test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "application",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/application/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 7,
            "startColumn": 9,
            "startLine": 7,
          },
        },
      },
    ],
    "message": {
      "text": "application test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "application",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/rendering/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 5,
            "startColumn": 9,
            "startLine": 5,
          },
        },
      },
    ],
    "message": {
      "text": "rendering test found using test",
    },
    "properties": {
      "testMethod": "test",
      "testType": "rendering",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/rendering/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 6,
            "startColumn": 9,
            "startLine": 6,
          },
        },
      },
    ],
    "message": {
      "text": "rendering test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "rendering",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/rendering/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 7,
            "startColumn": 9,
            "startLine": 7,
          },
        },
      },
    ],
    "message": {
      "text": "rendering test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "rendering",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/unit/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 5,
            "startColumn": 9,
            "startLine": 5,
          },
        },
      },
    ],
    "message": {
      "text": "unit test found using test",
    },
    "properties": {
      "testMethod": "test",
      "testType": "unit",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/unit/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 6,
            "startColumn": 9,
            "startLine": 6,
          },
        },
      },
    ],
    "message": {
      "text": "unit test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "unit",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
  {
    "kind": "informational",
    "level": "note",
    "locations": [
      {
        "physicalLocation": {
          "artifactLocation": {
            "uri": "addon/unit/tests/my-component-test.js",
          },
          "region": {
            "endColumn": 13,
            "endLine": 7,
            "startColumn": 9,
            "startLine": 7,
          },
        },
      },
    ],
    "message": {
      "text": "unit test found using skip",
    },
    "properties": {
      "testMethod": "skip",
      "testType": "unit",
    },
    "ruleId": "ember/ember-test-types",
    "ruleIndex": 0,
  },
]
`;
